stratum/hal/lib/tdi/CMakeLists.txt

```diff
@@ -0,0 +1,121 @@
+# CMake build file for Stratum hal tdi library.
+#
+# Copyright 2022 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+
+add_library(stratum_tdi_o OBJECT
+    tdi_action_profile_manager.cc
+    tdi_action_profile_manager.h
+    tdi_counter_manager.cc
+    tdi_counter_manager.h
+    tdi_id_mapper.cc
+    tdi_id_mapper.h
+    tdi_node.cc
+    tdi_node.h
+    tdi_packetio_manager.cc
+    tdi_packetio_manager.h
+    tdi_pipeline_utils.cc
+    tdi_pipeline_utils.h
+    tdi_pre_manager.cc
+    tdi_pre_manager.h
+    tdi_sde_action_profile.cc
+    tdi_sde_clone_session.cc
+    tdi_sde_common.h
+    tdi_sde_counter.cc
+    tdi_sde_helpers.cc
+    tdi_sde_helpers.h
+    tdi_sde_interface.h
+    tdi_sde_meter.cc
+    tdi_sde_multicast.cc
+    tdi_sde_register.cc
+    tdi_sde_table_data.cc
+    tdi_sde_table_entry.cc
+    tdi_sde_table_key.cc
+    tdi_sde_utils.h
+    tdi_sde_wrapper.cc
+    tdi_sde_wrapper.h
+    tdi_table_manager.cc
+    tdi_table_manager.h
+    utils.cc
+    utils.h
+)
+
+target_include_directories(stratum_tdi_o PRIVATE
+    ${STRATUM_INCLUDES}
+    ${SDE_INSTALL_DIR}/include
+)
+
+add_dependencies(stratum_tdi_o stratum_proto)
+
+if(TOFINO_TARGET)
+  add_subdirectory(tofino)
+elseif(DPDK_TARGET)
+  add_subdirectory(dpdk)
+endif()
+
+add_library(stratum_tdi
+    $<TARGET_OBJECTS:stratum_glue_o>
+    $<TARGET_OBJECTS:stratum_lib_o>
+    $<TARGET_OBJECTS:stratum_public_o>
+    $<TARGET_OBJECTS:stratum_hal_lib_common_o>
+    $<TARGET_OBJECTS:stratum_hal_lib_phal_o>
+    $<TARGET_OBJECTS:stratum_hal_lib_p4_o>
+    $<TARGET_OBJECTS:stratum_hal_lib_yang_o>
+    $<TARGET_OBJECTS:stratum_tdi_o>
+
+    $<$<BOOL:${TOFINO_TARGET}>:$<TARGET_OBJECTS:stratum_hal_lib_yang_subtree_interface_o>>
```

Comment:
This approach doesn't scale particularly well. It's a binary choice now, but it's likely to become a ternary choice when MEV enters into the picture.

It is apparently possible to distribute the commands that add to a library to different parts of the build.

If this is correct, each target subdirectory of hal/lib/tdi could use `target_sources` to add its own source files to `stratum_hal_lib_yang_o`.

It may get even messier than this when individual targets start wanting to suppress YANG nodes that don't apply to them. With luck, we'll have transitioned to a generated-code solution by then, but we can't count on that.

Reply:
We will never enable more than 1 target at the same time.
Having said that, using target_sources is an alternate option. One problem is I foresee is it will make it really hard to figure out who is building what. The above model keeps the object file within the same directory the source file is present. target_sources would break that model.
The current model I chose mirrors the bazel model where we we pick what we want to build with independent of where the source file sits.